gamemodes/terrortown/gamemode/client/cl_voice.lua

```diff
@@ -346,6 +371,116 @@ end
 VOICE.battery_max = 100
 VOICE.battery_min = 10
 
+---
+-- Scales a linear volume into a Power 4 value.
+-- @param number volume
+-- @realm client
+function VOICE.LinearToPower4(volume)
+	return math.Clamp( math.pow(volume, 4), 0, 1)
+end
+
+---
+-- Scales a linear volume into a Log value.
+-- @param number volume
+-- @realm client
+function VOICE.LinearToLog(volume)
+	local rolloff_cutoff = 0.1
+	local log_a = math.pow(1 / 10, 60 / 20)
+	local log_b = math.log(1 / log_a)
+
+	local vol = log_a * math.exp(log_b * volume)
+	if volume < rolloff_cutoff then
+		local log_rolloff = 10 * log_a * math.exp(log_b * rolloff_cutoff)
+		vol = volume * log_rolloff
+	end
+
+	return math.Clamp(vol, 0, 1)
+end
+
+---
+-- Passes along the input linear volume value.
+-- @param number volume
+-- @realm client
+function VOICE.LinearToLinear(volume)
+	return volume
+end
+
+VOICE.ScalingFunctions = {
+	power4 = VOICE.LinearToPower4,
+	log = VOICE.LinearToLog,
+	linear = VOICE.LinearToLinear,
+}
+
+VOICE.GetScalingFunctions = function()
+	local opts = {}
+	for mode in pairs(VOICE.ScalingFunctions) do
+		table.insert(opts, {
```

Comment:
at some point we decided to use absolute indexing when inserting into a table because it is waay faster. I guess this should be possible here as well?

Something like
```lua
table[#table + 1] = <new entry>
```